src/Views/.editorconfig

```
@@ -0,0 +1,10 @@
+; This file is for unifying the coding style for different editors and IDEs.
+; More information at http://editorconfig.org
+
+[*]
+charset = utf-8
+indent_size = 2
+indent_style = tab
```

```suggestion
indent_size = 4
indent_style = space
```

Let's keep this consistent with CI itself.

---

CI4 uses tabs in Views:
https://github.com/codeigniter4/CodeIgniter4/blob/develop/app/Views/welcome_message.php
https://github.com/codeigniter4/CodeIgniter4/blob/develop/app/Views/errors/html/error_404.php

---

Is there something that automates that? Or did that just never get changed when we made the switch to follow PSR-12? I don't know of any way to enforce it only in views and not other files, and that just gets confusing for people.

---

I think we did not change the indentation in Views, because these are not pure PHP source code.
In fact these are HTML files. So we still use tabs. And I think it is okay.

The `.editorconfig` enforces it only in this directory if users use `.editorconfig` aware IDEs.

---

I'm not sure if it was an oversight or desired behavior, but the switch to spaces was never applied to Views because they are all exempt in our CS Fixer. I have maintained this behavior in my own projects, not out of some intentional preference but just to remain consistent with the framework core.

---

I don't have strong preference for tab indentation. But I feel 2 space sized indentation is more common in HTML/CSS.

I also sent a PR to CI4 main repos. https://github.com/codeigniter4/CodeIgniter4/pull/6010

---

Wouldn't it be simpler just to keep everything the same?

---

@lonnieezell First, let's decide the rules for the main repository.
https://github.com/codeigniter4/CodeIgniter4/pull/6010